### PR TITLE
[SofaGui] Package SofaGui

### PR DESCRIPTION
**Pull request by fredroy**
* Usual cleaning (pragma, namespaces)
* Package SofaGuiCommon, SofaGuiQt, SofaHeadlessRecorder : cleaner deps handling

⚠ SofaGuiMain has been removed as it was doing nothing except GUI registrations. 
So it was deemed more convenient to move those registrations into SofaGui itself.
And people was misusing SofaGuiMain as SofaGui anyway: lots of projects were linking with SofaGuiMain, where it would have been more correct to link with SofaGuiCommon or SofaGuiQt.
The only relevant case to link SofaGuiMain was for applications wishing to run GUIs. (i.e runSofa, Modeler)

So now if you need to create a an application with a GUI using SofaGuiCommon/SofaGuiQt, you have to link with SofaGui instead of SofaGuiMain (and do initSofaGui() instead of initSofaMain() to register batch/qt/qglviewer/headlessrecorder modes)

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by damienmarchal** [comment]
Look good to me.

File: SofaGui/CMakeLists.txt [hunk 1]
```diff
@@ -1,50 +1,34 @@
...
-    message(WARNING "SOFAGUI_BUILD_TESTS can't be enabled without SOFA_BUILD_TESTS.")
-    set(SOFAGUI_BUILD_TESTS OFF CACHE BOOL "Compile the tests that need a GUI (like headlessRecorder_test)." FORCE)
-endif()
-
-### Sub-projects
-
-## SofaGuiCommon
+# SofaGuiCommon
 add_subdirectory(../modules/SofaGuiCommon ${CMAKE_CURRENT_BINARY_DIR}/SofaGuiCommon)
+set_target_properties(SofaGuiCommon PROPERTIES FOLDER SofaGui)
```
Comment (jnbrunet) [inline SofaGui/CMakeLists.txt]: Duplicated on line 29

File: modules/SofaGuiQt/CMakeLists.txt [hunk 2]
```diff
@@ -306,13 +306,18 @@ install(FILES "${CMAKE_BINARY_DIR}/etc/installed${PROJECT_NAME}.ini" DESTINATION
...
-sofa_add_targets_to_package(
-    PACKAGE_NAME SofaGui
+sofa_create_package_with_targets(
+    PACKAGE_NAME ${PROJECT_NAME}
+    PACKAGE_VERSION ${Sofa_VERSION}
     TARGETS ${PROJECT_NAME} AUTO_SET_TARGET_PROPERTIES
     INCLUDE_SOURCE_DIR "src"
-    INCLUDE_INSTALL_DIR "SofaGui"
+    INCLUDE_INSTALL_DIR ${PROJECT_NAME}
     )
```
Comment (jnbrunet) [inline modules/SofaGuiQt/CMakeLists.txt:316]: I would put SofaGuiQt and SofaHeadlessRecorder as relocatable, what do you think? We can keep SofaGuiCommon in the "kernel" (sofa/lib).

File: modules/SofaGuiQt/SofaGuiQtConfig.cmake.in [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+
+find_package(SofaBaseVisual QUIET REQUIRED)
+find_package(SofaLoader QUIET REQUIRED)
+find_package(SofaMiscForceField QUIET REQUIRED)
+find_package(SofaGuiCommon QUIET REQUIRED)
+
+if(SOFAGUIQT_HAVE_QTVIEWER)
+    find_package(Qt5 QUIET REQUIRED Core Gui OpenGL)
+    if(SOFAGUIQT_HAVE_QT5_CHARTS)
```
Comment (jnbrunet) [inline modules/SofaGuiQt/SofaGuiQtConfig.cmake.in:21]: Are the charts and the webengine really dependant on the **QtViewer**?

File: modules/SofaGuiQt/CMakeLists.txt (hunk 2 above)
Comment (fredroy) [reply modules/SofaGuiQt/CMakeLists.txt:316]: if it is just setting the relocatable flag, can do 😗

**Review by jnbrunet — APPROVED** [review]
Everything looks good to me. I added some small remarks/questions, but else it is ready to be merged whenever @fredroy feels it.

File: modules/SofaGuiQt/SofaGuiQtConfig.cmake.in (hunk 3 above)
Comment (fredroy) [reply modules/SofaGuiQt/SofaGuiQtConfig.cmake.in:21]: hum this is the current behavior in Sofa right now, so I did not really go further...

**Comment by damienmarchal** [comment]
So @fredroy we wait your signal :)

**Comment by fredroy** [comment]
> 
> 
> So @fredroy we wait your signal :)

I will do the duplicate and the relocatable flag, and it is good to go

**Pull request merged**